src/Microsoft.Health.SqlServer/Features/Schema/Manager/SchemaManagerDataStore.cs

```diff
@@ -7,21 +7,27 @@
 using System.Data;
 using System.Threading;
 using System.Threading.Tasks;
+using EnsureThat;
 using Microsoft.Data.SqlClient;
 using Microsoft.SqlServer.Management.Common;
 using Microsoft.SqlServer.Management.Smo;
 
-namespace SchemaManager
+namespace Microsoft.Health.SqlServer.Features.Schema.Manager
 {
-    public static class SchemaDataStore
+    public class SchemaManagerDataStore : ISchemaManagerDataStore
     {
-        public const string DeleteQuery = "DELETE FROM dbo.SchemaVersion WHERE Version = @version AND Status = @status";
-        public const string Failed = "failed";
-        public const string Completed = "completed";
+        private ISqlConnectionFactory _sqlConnectionFactory;
 
-        public static async Task ExecuteScriptAndCompleteSchemaVersionAsync(string connectionString, string script, int version, CancellationToken cancellationToken)
+        public SchemaManagerDataStore(ISqlConnectionFactory sqlConnectionFactory)
         {
-            using (var connection = new SqlConnection(connectionString))
+            EnsureArg.IsNotNull(sqlConnectionFactory);
+
+            _sqlConnectionFactory = sqlConnectionFactory;
+        }
+
+        public async Task ExecuteScriptAndCompleteSchemaVersionAsync(string script, int version, CancellationToken cancellationToken)
```

Comment:
Could you add parameter validation to public methods?

Reply:
Done